roles/linux/opensearch/tasks/security.yml

```diff
@@ -137,21 +225,59 @@
   environment:
     JAVA_HOME: "{{ os_home }}/jdk"
   run_once: true
-  when: configuration.changed
+  when: configuration.changed or iac_enable
+
+- name: Security Plugin configuration | Check that the files/internal_users.yml exists
+  stat:
+    path: files/internal_users.yml
+  register: custom_users_result
+  delegate_to: localhost
+  run_once: true
+
+- name: Security Plugin configuration | Check for a custom configuration for internal users and hash passwords for them
+  block:
+
+    - name: Security Plugin configuration | Load custom internal users configuration
+      include_vars:
+        file: files/internal_users.yml
+        name: custom_users
+      run_once: true
+
+    - name: Security Plugin configuration | Filter service keys from the list of users
+      set_fact:
+        custom_users_filtered: '{{ custom_users|dict2items|rejectattr("key", "equalto", "_meta")|list|items2dict }}'
+
+#    - name: Security Plugin configuration | debug
+#      debug:
+#        msg: "{{ item }} password: {{ lookup('vars', item + '_password') }}"
+#      with_items: "{{ custom_users_filtered }}"
+#      run_once: true
+
+    - name: Security Plugin configuration | Set passwords for all users from custom config
+      shell: >
+        sed -i '/hash: / s,{{ lookup('vars', item + '_password') }},'$(bash {{ os_sec_plugin_tools_path }}/hash.sh -p {{ lookup('vars', item + '_password') }} | tail -1)','
+        {{ os_sec_plugin_conf_path }}/internal_users.yml
+      environment:
+        JAVA_HOME: "{{ os_home }}/jdk"
+      run_once: true
+      when: configuration.changed or copy_custom_security_configs
```

Comment:
@ssi444 so can you add some details on when this block will be triggered, when a password done by user using `admin_password` or similar?

Reply:
In the file README.md in the "Prerequisite" section, I added a description of how to set passwords for custom users (line 92-94)
https://github.com/ssi444/opensearch-project_ansible-playbook/blob/4b4bf51a01ac2e6db9a42c78e18914dab695063d/README.md?plain=1#L92

Reply:
That comes for copy_custom_security_configs I guess, just wanted to know about configuration.changed, when an admin user changed ?